File: javascript-ads/src/data-structures/singly-linked-list/SinglyLinkedList.js
const { Node } = require('./Node');

class SinglyLinkedList {
    constructor() {
        this.head = null;
        this.count = 0;
    }

    any() {
        return this.head !== null;
    }

    addFront(value) {
        const node = new Node(value);
        node.next = this.head;

        this.head = node;

        this.count++;
    }

    removeFront() {
        if (!this.any()) {
            return null;
        }

        const removedNode = this.head;
        this.head = removedNode.next;
        removedNode.next = null;

        this.count--;

        return removedNode;
    }

    toArray() {
        const nodes = [];

        let currentNode = this.head;

        while (currentNode != null) {
            nodes.push(currentNode.value);
            currentNode = currentNode.next;
        }

        return nodes;
    }

    toString() {
        const nodes = this.toArray();
        return `SinglyLinkedList [ ${nodes.join(' -> ')} ]`;
    }
}

module.exports = { SinglyLinkedList };

File: csharp-ads/src/ADS.DataStructures/SinglyLinkedList/SinglyLinkedList.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ADS.DataStructures.SinglyLinkedList
{
    public sealed class SinglyLinkedList<T> where T : class
    {
        public long Count { get; private set; }

        public Node<T> Head { get; private set; }

        public bool Any()
        {
            return Head != null;
        }

        public void AddFront(T value)
        {
            var node = new Node<T>(value);
            node.Next = Head;

            Head = node;

            Count++;
        }

        public Node<T> RemoveFront()
        {
            if (!Any())
            {
                return null;
            }

            var removedNode = Head;
            Head = removedNode.Next;
            removedNode.Next = null;
            Count--;

            return removedNode;
        }

        public IEnumerable<T> ToEnumerable()
        {
            var node = Head;

            while (node != null)
            {
                yield return node.Value;
                node = node.Next;
            }
        }

        public List<T> ToList()
        {
            return this.ToEnumerable().ToList();
        }

        public override string ToString()
        {
            var nodes = string.Join(" -> ", ToList());
            var className = nameof(SinglyLinkedList<T>);

            return $"{className} [ {nodes} ]";
        }
    }
}
